connection.js

```diff
@@ -67,6 +72,23 @@ function connect(id, password, roomName) {
         connection.addEventListener(
             ConnectionEvents.CONNECTION_FAILED, handleConnectionFailed
         );
+        connection.addEventListener(
+            ConnectionEvents.CONNECTION_FAILED, connectionFailedHandler);
+
+        function connectionFailedHandler (error, errMsg) {
+            APP.store.dispatch(connectionFailed(connection, error, errMsg));
+
+            switch (error) {
+            case ConnectionErrors.CONNECTION_DROPPED_ERROR:
+            case ConnectionErrors.OTHER_ERROR:
+            case ConnectionErrors.SERVER_ERROR: {
```

Comment:
Saúl commented on the fact that the 2 `CONNECTION_FAILED` handlers look almost duplicate and I understand how it may seem complex to reduce duplication. However, we have to agree that we have a `switch (error)` with the same group of 3 `case`s in 3 files just in this PR. Obviously, there's something special about these 3 `ConnectionErrors`. What's not obvious (to me) is what that special thing is. So it would've been both a duplication reduction and a clarification if the testing for these 3 `cases` was in a properly named function. It could've even been a method on `ConnectionErrors`.